contrib/natss/pkg/controller/clusterchannelprovisioner/controller.go

```diff
@@ -27,25 +25,20 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 
 	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
-	"github.com/knative/eventing/pkg/utils"
 	corev1 "k8s.io/api/core/v1"
 )
 
 const (
-	// NATSS
-	ClusterId    = "knative-nats-streaming"
-	NatssUrlTmpl = "nats://nats-streaming.natss.svc.%s:4222"
-	clientId     = "knative-natss-controller"
+	clientID = "knative-natss-controller"
 	// controllerAgentName is the string used by this controller to identify itself when creating events.
 	controllerAgentName = "natss-provisioner-controller"
 )
 
 // ProvideController returns a flow controller.
-func ProvideController(mgr manager.Manager, logger *zap.Logger) (controller.Controller, error) {
+func ProvideController(mgr manager.Manager, defaultNatssURL, defaultClusterID string, logger *zap.Logger) (controller.Controller, error) {
```

Comment:
Do we ever connect to anything but these natssURL and clusterID? If not, I would remove 'default'.